packages/components/inputs/async-creatable-select-input/src/async-creatable-select-input.tsx

```diff
@@ -237,173 +102,377 @@ AsyncCreatableSelectInput.propTypes = {
    */
-  backspaceRemovesValue: PropTypes.bool,
+  backspaceRemovesValue?: ReactSelectProps['backspaceRemovesValue'];
   /**
    * Map of components to overwrite the default ones, see [what components you can override](https://react-select.com/components)
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  components: PropTypes.objectOf(PropTypes.func),
+  components?: ReactSelectProps['components'];
   /**
    * Custom method to filter whether an option should be displayed in the menu
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  filterOption: PropTypes.func,
+  filterOption?: ReactSelectProps['filterOption'];
   // This forwarded as react-select's "inputId"
   /**
    * The id of the search input
    */
-  id: PropTypes.string,
+  id?: string;
   // This is forwarded as react-select's "id"
-  inputValue: PropTypes.string,
+  /**
+   * The value of the search input
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
+   */
+  inputValue?: ReactSelectProps['inputValue'];
   /**
    * The id to set on the SelectContainer component
    */
-  containerId: PropTypes.string,
+  containerId?: string;
   /**
    * Is the select value clearable
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  isClearable: PropTypes.bool,
+  isClearable?: ReactSelectProps['isClearable'];
   /**
    * Is the select disabled
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  isDisabled: PropTypes.bool,
+  isDisabled?: ReactSelectProps['isDisabled'];
   /**
    * Override the built-in logic to detect whether an option is disabled
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  isOptionDisabled: PropTypes.func,
+  isOptionDisabled?: ReactSelectProps['isOptionDisabled'];
   /**
    * Support multiple selected options
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  isMulti: PropTypes.bool,
+  isMulti?: ReactSelectProps['isMulti'];
   /**
    * Whether to enable search functionality
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  isSearchable: PropTypes.bool,
+  isSearchable?: ReactSelectProps['isSearchable'];
   /**
    * Maximum height of the menu before scrolling
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  maxMenuHeight: PropTypes.number,
+  maxMenuHeight?: ReactSelectProps['maxMenuHeight'];
   /**
    * Dom element to portal the select menu to
+   * <br>
+   * [Props from React select was used](https://react-select.com/props)
    */
-  menuPortalTarget: PropTypes.instanceOf(SafeHTMLElement),
+  menuPortalTarget?: ReactSelectProps['menuPortalTarget'];
   /**
    * z-index value for the menu portal
    */
-  menuPortalZIndex: PropTypes.number.isRequired,
+  menuPortalZIndex?: number;
```

Comment:
menuPortalZIndex is required
```suggestion
  menuPortalZIndex: number;
```

Reply:
Thanks.